include/gatekeeper_gk.h

```diff
@@ -81,7 +99,21 @@ struct gk_nexthop {
 	struct ipaddr ip_addr;
 
 	/* The cached Ethernet header. */
-	struct ether_cache eth_cache;
+	struct ether_cache *eth_cache;
+};
```

Comment:
It seems that `struct gk_nexthop` is only needed to define `struct gk_fib`. If so, please move the definition of `struct gk_nexthop` into `struct gk_fib`; it helps readability.